src/chunkformatter.js

```diff
@@ -0,0 +1,311 @@
+    'A reset should have no data',
+    chunk
+  );
+};
+/**
+ * Validates state for new row.
+ * @private
+ * @param {chunk} chunk chunk to validate
+ */
+ChunkFormatter.prototype.validateNewRow = function(chunk) {
+  const row = this.row;
+  const prevRowKey = this.prevRowKey;
+  const newRowKey = Mutation.convertFromBytes(chunk.rowKey);
+  this.isError(
+    typeof row.key !== 'undefined',
+    'A new row cannot have existing state',
+    chunk
+  );
+  this.isError(
+    typeof chunk.rowKey === 'undefined' || chunk.rowKey === '',
+    'A row key must be set',
+    chunk
+  );
+  this.isError(chunk.resetRow, 'A new row cannot be reset', chunk);
+  this.isError(
+    prevRowKey === newRowKey,
+    'A commit happened but the same key followed',
+    chunk
+  );
+  this.isError(!chunk.familyName, 'A family must be set', chunk);
+  this.isError(!chunk.qualifier, 'A column qualifier must be set', chunk);
+  this.validateValueSizeAndCommitRow(chunk);
+};
+/**
+ * Validates state for rowInProgress
+ * @private
+ * @param {chunk} chunk chunk to validate
+ */
+ChunkFormatter.prototype.validateRowInProgress = function(chunk) {
+  const row = this.row;
+  this.validateResetRow(chunk);
+  const newRowKey = Mutation.convertFromBytes(chunk.rowKey);
+  this.isError(
+    chunk.rowKey && newRowKey !== row.key,
+    'A commit is required between row keys',
+    chunk
+  );
+  this.validateValueSizeAndCommitRow(chunk);
+  this.isError(
+    chunk.familyName && !chunk.qualifier,
+    'A qualifier must be specified',
+    chunk
+  );
+};
+/**
+ * Validates chunk for cellInProgress state.
+ * @private
+ * @param {chunk} chunk chunk to validate
+ */
+ChunkFormatter.prototype.validateCellInProgress = function(chunk) {
+  this.validateResetRow(chunk);
+  this.validateValueSizeAndCommitRow(chunk);
+};
+/**
+ * Moves to next state in processing.
+ * @private
+ * @param {chunk} chunk chunk in process
+ * @param {*} callback callback to call with row as and when generates
+ */
+ChunkFormatter.prototype.moveToNextState = function(chunk, callback) {
+  const row = this.row;
+  if (chunk.commitRow) {
+    callback(null, row);
+    this.commit();
+  } else {
+    if (chunk.valueSize > 0) {
+      this.state = RowStateEnum.CELL_IN_PROGRESS;
+    } else {
+      this.state = RowStateEnum.ROW_IN_PROGRESS;
+    }
+  }
+};
+/**
+ * Process chunk when in NEW_ROW state.
+ * @private
+ * @param {chunks} chunk chunk to process
+ * @param {options} options options
+ * @param {callback} callback callback to call with row as and when generates
+ */
+ChunkFormatter.prototype.newRow = function(chunk, options, callback) {
+  const newRowKey = Mutation.convertFromBytes(chunk.rowKey);
+  this.validateNewRow(chunk);
+  const row = this.row;
+  row.key = newRowKey;
+  row.data = {};
+  this.family = row.data[chunk.familyName.value] = {};
+  const qualifierName = Mutation.convertFromBytes(chunk.qualifier.value);
+  this.qualifiers = this.family[qualifierName] = [];
+  this.qualifier = {
```

Comment:
`this.qualifier` should probably be named `this.cellInProgress`.

Reply:
this.row, this.family and this.qualifier in the ChunkFormatter instance is in progress row, family & qualifier. Also there I named a method cellInProgress to process CELL_IN_PROGRESS state, It will be confusing with that name. If we want to make it explicit, I can rename state processing method to processNewRow, processRowInProgress, processCellInProgress & rename state holder to rowInProgress, familyInProgress & cellInProgress. Let me know.